tests/pico/stock_nightly_no_std/src/pico_rt.rs

```diff
@@ -0,0 +1,136 @@
+use core::alloc::{GlobalAlloc, Layout};
+// Import user `main` function
+use crate::main;
+
+// 1. Init global pointer (GP). It's used to optimize jumps by linker. Linker can change jumping from PC(Program Counter) based to GP based.
+// 2. Init stack pointer to the value STACK_TOP. It's stored in sp register.
+// 3. Call __start function defined below.
+// `__global_pointer$` is set by the linker. Its value depends on linker optimization. https://www.programmersought.com/article/77722901592/
+core::arch::global_asm!(
+    r#"
+.section .text._start;
+.globl _start;
+_start:
+    .option push;
+    .option norelax;
+    la gp, __global_pointer$;
+    .option pop;
+    la sp, {0}
+    lw sp, 0(sp)
+    call __start;
+"#,
+    sym STACK_TOP
+);
+
+static STACK_TOP: u32 = 0x0020_0400;
+
+// 1. Call `main` user function
+// 2. Call system halt environment function. It's defined by sp1 vm.
+#[unsafe(no_mangle)]
+fn __start(_argc: isize, _argv: *const *const u8) -> isize {
+    main();
+
+    syscall_halt(0);
+    unreachable!()
+}
+
+/// Halts the program with the given exit code.
+/// According to https://github.com/brevis-network/pico/blob/v1.1.7/sdk/sdk/src/riscv_ecalls/halt.rs#L15
+/// TODO: Check what `coprocessor` feature does and integrate here.
```

Comment:
Can you create an issue for this?

Reply:
https://linear.app/zkevm/issue/ZKE-301/update-pico-rtrs-to-make-it-fully-compatible-with-pico-implementation